### PR TITLE
Better handling for TS generics

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Fixes https://github.com/withastro/astro/issues/3793
- Better handling for TS generics inside of expressions

## Testing

Test added

## Docs

Bug fix only

### CODE REVIEW

File: internal/token_test.go [hunk 1]
```diff
@@ -399,6 +399,21 @@ func TestBasic(t *testing.T) {
...
+		{
+			"typescript generic II",
+			`<ul>{items.map((item: Item<Checkbox>)) => <Checkbox>{item.checked}</Checkbox>)}</ul>`,
+			[]TokenType{StartTagToken, StartExpressionToken, TextToken, TextToken, TextToken, StartTagToken, StartExpressionToken, TextToken, EndExpressionToken, EndTagToken, TextToken, EndExpressionToken, EndTagToken},
+		},
+		{
+			"typescript generic III",
+			`<ul>{items.map((item: Item<Checkbox>)) => <Checkbox>{item.checked}</Checkbox>)}</ul>`,
+			[]TokenType{StartTagToken, StartExpressionToken, TextToken, TextToken, TextToken, StartTagToken, StartExpressionToken, TextToken, EndExpressionToken, EndTagToken, TextToken, EndExpressionToken, EndTagToken},
+		},
```
Comment (Princesseuh) [inline internal/token_test.go:416]: Not sure if intended, but those two tests are the same

File: internal/token.go [hunk 2]
```diff
@@ -1002,6 +1002,26 @@ func (z *Tokenizer) readStartTag() TokenType {
 		return SelfClosingTagToken
 	}
 
+	// Handle TypeScript Generics
+	if len(z.expressionElementStack) > 0 && len(z.expressionElementStack[len(z.expressionElementStack)-1]) == 0 {
+		if z.prevToken.Type == TextToken {
+			tag := z.buf[z.data.Start:z.data.End]
+			a := atom.Lookup(tag)
+			fmt.Println(string(tag))
```
Comment (Princesseuh) [inline internal/token.go:1010]: Leftover Go `console.log`, I believe?

**Review by Princesseuh — APPROVED** [review]
Trying to learn reviewing compiler PRs, so take this review with a grain of salt, but apart from the two comments, LGTM 😄

**Comment by nirus** [comment]
seems the issue still exists!

**Pull request merged**